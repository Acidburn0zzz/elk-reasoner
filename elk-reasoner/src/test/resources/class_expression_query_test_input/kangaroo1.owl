Prefix(:=<owlapi:ontology#ont9#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)


Ontology(<owlapi:ontology#ont9>

Declaration(Class(<http://www.test.com/schema#Animal>))
Declaration(Class(<http://www.test.com/schema#Beast>))
Declaration(Class(<http://www.test.com/schema#Human>))
Declaration(Class(<http://www.test.com/schema#Irrational>))
Declaration(Class(<http://www.test.com/schema#Kangaroo>))
Declaration(Class(<http://www.test.com/schema#KangarooInfant>))
Declaration(Class(<http://www.test.com/schema#Parent>))
Declaration(Class(<http://www.test.com/schema#Pouch>))
Declaration(Class(<http://www.test.com/schema#Rational>))
Declaration(ObjectProperty(<http://www.test.com/schema#has-child>))
Declaration(ObjectProperty(<http://www.test.com/schema#lives-in>))
############################
#   Object Properties
############################

# Object Property: <http://www.test.com/schema#has-child> (<http://www.test.com/schema#has-child>)

ObjectPropertyDomain(<http://www.test.com/schema#has-child> <http://www.test.com/schema#Human>)
ObjectPropertyRange(<http://www.test.com/schema#has-child> <http://www.test.com/schema#Human>)


############################
#   Classes
############################

# Class: <http://www.test.com/schema#Beast> (<http://www.test.com/schema#Beast>)

SubClassOf(<http://www.test.com/schema#Beast> ObjectIntersectionOf(<http://www.test.com/schema#Animal> <http://www.test.com/schema#Irrational>))

# Class: <http://www.test.com/schema#Human> (<http://www.test.com/schema#Human>)

SubClassOf(<http://www.test.com/schema#Human> ObjectIntersectionOf(<http://www.test.com/schema#Animal> <http://www.test.com/schema#Rational>))

# Class: <http://www.test.com/schema#Irrational> (<http://www.test.com/schema#Irrational>)

DisjointClasses(<http://www.test.com/schema#Irrational> <http://www.test.com/schema#Rational>)

# Class: <http://www.test.com/schema#Kangaroo> (<http://www.test.com/schema#Kangaroo>)

SubClassOf(<http://www.test.com/schema#Kangaroo> <http://www.test.com/schema#Beast>)

# Class: <http://www.test.com/schema#KangarooInfant> (<http://www.test.com/schema#KangarooInfant>)

SubClassOf(<http://www.test.com/schema#KangarooInfant> ObjectIntersectionOf(<http://www.test.com/schema#Kangaroo> ObjectSomeValuesFrom(<http://www.test.com/schema#lives-in> <http://www.test.com/schema#Pouch>)))

# Class: <http://www.test.com/schema#Parent> (<http://www.test.com/schema#Parent>)

EquivalentClasses(<http://www.test.com/schema#Parent> ObjectSomeValuesFrom(<http://www.test.com/schema#has-child> owl:Thing))


)