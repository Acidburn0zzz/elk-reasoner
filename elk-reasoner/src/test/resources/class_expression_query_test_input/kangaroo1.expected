Prefix(:=<owlapi:ontology#ont10#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)


Ontology(<owlapi:ontology#ont10>

Declaration(Class(<http://www.test.com/schema#Kangaroo>))
Declaration(Class(<http://www.test.com/schema#KangarooInfant>))
Declaration(Class(<http://www.test.com/schema#Parent>))
Declaration(Class(<http://www.test.com/schema#Pouch>))
Declaration(ObjectProperty(<http://www.test.com/schema#has-body-part>))
Declaration(ObjectProperty(<http://www.test.com/schema#has-child>))

############################
#   Classes
############################

# Class: owl:Nothing (owl:Nothing)

EquivalentClasses(owl:Nothing ObjectIntersectionOf(<http://www.test.com/schema#Kangaroo> ObjectSomeValuesFrom(<http://www.test.com/schema#has-body-part> <http://www.test.com/schema#Pouch>) ObjectSomeValuesFrom(<http://www.test.com/schema#has-child> <http://www.test.com/schema#KangarooInfant>)))


SubClassOf(ObjectIntersectionOf(<http://www.test.com/schema#Kangaroo> ObjectSomeValuesFrom(<http://www.test.com/schema#has-body-part> <http://www.test.com/schema#Pouch>) ObjectSomeValuesFrom(<http://www.test.com/schema#has-child> <http://www.test.com/schema#KangarooInfant>)) <http://www.test.com/schema#KangarooInfant>)
SubClassOf(ObjectIntersectionOf(<http://www.test.com/schema#Kangaroo> ObjectSomeValuesFrom(<http://www.test.com/schema#has-body-part> <http://www.test.com/schema#Pouch>) ObjectSomeValuesFrom(<http://www.test.com/schema#has-child> <http://www.test.com/schema#KangarooInfant>)) <http://www.test.com/schema#Parent>)
SubClassOf(ObjectIntersectionOf(<http://www.test.com/schema#Kangaroo> ObjectSomeValuesFrom(<http://www.test.com/schema#has-body-part> <http://www.test.com/schema#Pouch>) ObjectSomeValuesFrom(<http://www.test.com/schema#has-child> <http://www.test.com/schema#KangarooInfant>)) <http://www.test.com/schema#Pouch>)
)